x-pack/plugins/uptime/public/components/monitor/ping_list/columns/ping_timestamp.tsx

```diff
@@ -140,17 +150,36 @@ export const PingTimestamp = ({ timestamp, ping }: Props) => {
   );
 
   return (
-    <StepDiv ref={intersectionRef}>
+    <StepDiv
+      onMouseEnter={() => setIsImagePopoverOpen(true)}
+      onMouseLeave={() => setIsImagePopoverOpen(false)}
+      ref={intersectionRef}
+    >
       {imgSrc ? (
-        <StepImage
-          allowFullScreen={true}
-          size="s"
-          hasShadow
-          caption={ImageCaption}
-          alt={captionContent}
-          url={imgSrc}
-          data-test-subj="pingTimestampImage"
-        />
+        <EuiPopover
+          anchorPosition="rightCenter"
+          button={
+            <StepImage
+              allowFullScreen={true}
+              alt={captionContent}
+              caption={ImageCaption}
+              data-test-subj="pingTimestampImage"
+              hasShadow
+              url={imgSrc}
+              size="s"
+            />
+          }
+          closePopover={() => setIsImagePopoverOpen(false)}
+          isOpen={isImagePopoverOpen}
+        >
+          <EuiImage
+            alt={i18n.translate('xpack.uptime.synthetics.thumbnail.fullSize.alt', {
+              defaultMessage: `A full-size screenshot for this journey step's thumbnail.`,
+            })}
+            url={imgSrc}
+            style={{ height: POPOVER_IMG_HEIGHT, width: POPOVER_IMG_WIDTH, objectFit: 'contain' }}
+          />
+        </EuiPopover>
```

Comment:
seems like this ping_timestamp component has become bulky, i feel it deserver some asbtraction.

WDYT?

Reply:
Sure I can refactor. This code was somewhat new to me as it was contributed in my absence, so I didn't want to do any significant restructuring. I'm happy to try to break it up a bit though!